sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -326,30 +236,99 @@ export class ApplicationStudentsController extends BaseController {
-    } catch (error) {
-      switch (error.name) {
-        case APPLICATION_NOT_FOUND:
-          throw new NotFoundException(error.message);
-        case APPLICATION_NOT_VALID:
-        case INVALID_OPERATION_IN_THE_CURRENT_STATUS:
-        case STUDY_DATE_OVERLAP_ERROR:
-        case INSTITUTION_LOCATION_NOT_VALID:
-        case OFFERING_NOT_VALID:
-          throw new UnprocessableEntityException(
-            new ApiProcessError(error.message, error.name),
-          );
-        case ASSESSMENT_INVALID_OPERATION_IN_THE_CURRENT_STATE:
-          throw new UnprocessableEntityException(error.message);
-        default:
-          // TODO: add logger.
-          throw new InternalServerErrorException(
-            "Unexpected error while submitting the application.",
-          );
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError) {
+        switch (error.name) {
+          case APPLICATION_NOT_FOUND:
+            throw new NotFoundException(error.message);
+          case INVALID_OPERATION_IN_THE_CURRENT_STATUS:
+          case STUDY_DATE_OVERLAP_ERROR:
+          case INSTITUTION_LOCATION_NOT_VALID:
+          case OFFERING_NOT_VALID:
+            throw new UnprocessableEntityException(
+              new ApiProcessError(error.message, error.name),
+            );
+          case ASSESSMENT_INVALID_OPERATION_IN_THE_CURRENT_STATE:
+            throw new UnprocessableEntityException(error.message);
+        }
       }
+      throw error;
+    }
+  }
+
+  /**
+   * Starts a change request for an existing student application.
+   * A new application will be created in edited status and will be assessed by the Ministry.
+   * @param payload complete application payload with all the data that should be considered as a new
+   * application, if approved by the Ministry.
+   * @param applicationId application ID to be changed.
+   * @returns application ID of the created application that represents the change request.
+   */
+  @CheckSinValidation()
+  @Post(":applicationId/change-request")
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Program year is not active " +
+      "or selected offering ID is invalid " +
+      "or the education program is not active " +
+      "or the education program is expired " +
+      "or invalid offering intensity " +
+      "or an application change request is already in progress " +
+      "or application is not in the correct status to be submitted " +
+      "or change request has a different offering from its original submission" +
+      "or change request has a different location from its original submission.",
+  })
+  @ApiBadRequestResponse({
+    description:
+      "Form validation failed or offering intensity type is invalid.",
+  })
+  @ApiNotFoundResponse({
+    description:
+      "Application not found or it is not in the correct status to be changed.",
+  })
+  @ApiForbiddenResponse({
+    description: "The student has a restriction on his account.",
```

Comment:
Minor. `The student has a restriction.` would be sufficient and more generic.

Reply:
It was changed from the current submission that state "You have a restriction on your account.".
I believe the "The student has a restriction on his account." is generic and informative enough 😉